app/src/main/java/com/github/se/cyrcle/ui/profile/ProfileScreen.kt

```diff
@@ -1,192 +1,295 @@
+import androidx.compose.foundation.background
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.*
 import androidx.compose.foundation.lazy.LazyRow
 import androidx.compose.foundation.lazy.itemsIndexed
 import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material.icons.filled.Star
 import androidx.compose.material3.*
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
-import androidx.compose.ui.text.input.TextFieldValue
 import androidx.compose.ui.unit.dp
 import coil.compose.rememberAsyncImagePainter
 import coil.request.ImageRequest
 import coil.transform.CircleCropTransformation
-import com.github.se.cyrcle.model.user.User
+import com.github.se.cyrcle.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Route
 import com.github.se.cyrcle.ui.theme.ColorLevel
-import com.github.se.cyrcle.ui.theme.getButtonColors
+import com.github.se.cyrcle.ui.theme.atoms.Button
+import com.github.se.cyrcle.ui.theme.atoms.InputText
+import com.github.se.cyrcle.ui.theme.molecules.BottomNavigationBar
 
 @Composable
-fun ProfileScreen(user: User) {
+fun ProfileScreen(navigationActions: NavigationActions) {
   var isEditing by remember { mutableStateOf(false) }
-  var firstName by remember { mutableStateOf(TextFieldValue(user.firstName)) }
-  var lastName by remember { mutableStateOf(TextFieldValue(user.lastName)) }
-  var username by remember { mutableStateOf(TextFieldValue(user.username)) }
-
-  var favoriteParkings by remember { mutableStateOf(user.favoriteParkings.toMutableList()) }
-  val parkingFavorites = remember {
-    mutableStateListOf(*List(user.favoriteParkings.size) { true }.toTypedArray())
-  }
-
-  var profilePictureUrl by remember { mutableStateOf(user.profilePictureUrl) }
-  val defaultImageUrl =
-      "https://th.bing.com/th/id/R.aee6adef085a77dfa4708f3fd4a1ffb5?rik=nj%2buNqgLIJU0JQ&pid=ImgRaw&r=0"
-  val filledStarImageUrl =
-      "https://th.bing.com/th/id/R.f493b7a899f0c38194622d00ca70a7ff?rik=MvAc4%2fM5MeISxA&pid=ImgRaw&r=0"
-  val emptyStarImageUrl =
-      "https://cdn0.iconfinder.com/data/icons/thin-voting-awards/57/thin-232_star_favorite_add-1024.png"
+  var firstName by remember { mutableStateOf("John") }
+  var lastName by remember { mutableStateOf("Doe") }
+  var username by remember { mutableStateOf("johndoe") }
+  var profilePictureUrl by remember { mutableStateOf("") }
+  var favoriteParkings by remember { mutableStateOf(ArrayList<String>()) }
```

Comment:
It would be preferable to use a general list rather than arrayList. This also implies changing some other lines like 238, 239 and other places where ArrayList is specified.